test/hex_web/rate_limit_test.exs

```diff
@@ -0,0 +1,63 @@
+defmodule HexWeb.RateLimitTest do
+  use ExUnit.Case, async: false
+  use Plug.Test
+
+  defmodule Hello do
+    # need to use Phoenix.Controller because RateLimit.Plug uses ErrorView
+    # which in turn requres `plug :accepts`
+    use Phoenix.Controller
+
+    plug :accepts, ~w(json)
+    plug HexWeb.RateLimit.Plug
+
+    def index(conn, _params) do
+      send_resp(conn, 200, "Hello, World!")
+    end
+  end
+
+  test "allows requests when limit is not exceeded" do
+    conn = request({0, 0, 0, 0})
+    assert conn.status == 200
+    assert conn.resp_body == "Hello, World!"
+    assert %{"x-ratelimit-limit" => "100", "x-ratelimit-remaining" => "99"} = headers(conn)
+  end
+
+  test "halts requests when limit is exceeded" do
+    Enum.each(99..0, fn i ->
```

Comment:
not the best way to test it... Any other ideas? I thought about modifying `HexWeb.RateLimit` GenServer accept custom configuration that could be just used for this test but I couldn't get that to work